cmd/thanos/main.go

```diff
@@ -227,6 +231,37 @@ func defaultGRPCServerOpts(logger log.Logger, reg *prometheus.Registry, tracer o
 			grpc_recovery.StreamServerInterceptor(grpc_recovery.WithRecoveryHandler(grpcPanicRecoveryHandler)),
 		),
 	}
+
+	if insecure {
+		return opts, nil
+	}
+
+	tlsCfg := &tls.Config{}
```

Comment:
Worth to move this to separate function? And same for client grpc Opts?

Reply:
Not sure on the intention here, assuming all is well for now.